tests/unittests/regression/test_pearson.py

```diff
@@ -84,13 +112,17 @@ def test_pearson_corrcoef(self, preds, target, compute_on_cpu, ddp):
             metric_args={"num_outputs": num_outputs, "compute_on_cpu": compute_on_cpu},
         )
 
-    def test_pearson_corrcoef_functional(self, preds, target):
+    def test_pearson_corrcoef_functional(self, preds, target, metric_args):
         """Test functional implementation of metric."""
         self.run_functional_metric_test(
-            preds=preds, target=target, metric_functional=pearson_corrcoef, reference_metric=_scipy_pearson
+            preds=preds,
+            target=target,
+            metric_functional=pearson_corrcoef,
+            reference_metric=partial(_ref_metric, metric_args=metric_args),
+            metric_args=metric_args,
         )
 
-    def test_pearson_corrcoef_differentiability(self, preds, target):
+    def test_pearson_corrcoef_differentiability(self, preds, target, metric_args):
```

Comment:
if the new argument is not relevant to all the methods, lets consider moving it from parametrizing the class to just the relevant methods

Reply:
I will just parameterize `test_pearson_corrcoef` and `test_pearson_corrcoef_functional` methods with

```
@pytest.mark.parametrize(
    "metric_args",
    [({}), ({"weights": torch.rand(NUM_BATCHES, BATCH_SIZE)})],
)
```